pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -270,6 +274,11 @@ private BrokerResponseNative handleSQLRequest(long requestId, String query, Json
     if (_enableQueryLimitOverride) {
       handleQueryLimitOverride(pinotQuery, _queryResponseLimit);
     }
+    if (_enableSegmentPartitionedDistinctCountOverride
+        || QueryOptionsUtils.enableSegmentPartitionedDistinctCountOverride(pinotQuery.getQueryOptions())) {
+      handleSegmentPartitionedDistinctCountOverride(pinotQuery,
```

Comment:
This need to be done as a query optimizer. You won't be able to get the table config with the raw table name

Reply:
Actually I feel broker side is the right place.
E.g. if a hybrid table has configured this only for offline but not realtime  table, then we cannot perform the override.

Reply:
Yes, and that's the reason why I suggest modeling it as a query optimizer. Query optimizer is at broker, where we already have access to the table config